awesomemotive/all-in-one-seo-pack
# Sitemap priority/frequency for individual posts/terms

Issue #3122 & #2730

## Proposed changes

Adds support for overriding the global or default priority/frequency values for individual posts and taxonomy terms.

Also fixes a bug where taxonomies weren't able to fetch their default priority/frequency values, refactors some code and removes a few unused functions.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Improves existing code

## Checklist


## Testing instructions

1. Download this PR, build Pro with the latest 3.3.5 version, install & activate it and then add a few different CPTs that are assigned to a few custom taxonomies (WooCommerce Product Categories for instance).
2. Go to the Edit Post screen and override the priority/frequency. Then check the sitemap and confirm these are the values that are displayed. Do the same for the custom taxonomies.
3. Go to the XML Sitemap module and override the default values with a different value. At this point, the individual priority/frequency values should still take precedence. If you remove these by setting the fields to "Do Not Override", the plugin should fallback on the global values.
4. Remove the global values and confirm that the default values are used (check code snippets below to see what these are). Also check this for taxonomies. Previously, taxonomies displayed two blank fields. They should now be populated with the default values (#2730).
5. Make sure the tooltips are populated with the help text in the metabox.

`public function get_default_priority() {
			$defaults = array(
				'homepage'   => '1.0',
				'blog'       => '0.9',
				'sitemap'    => '0.8',
				'post'       => '0.7',
				'archive'    => '0.5',
				'author'     => '0.3',
				'taxonomies' => '0.3',
			);`

`		public function get_default_frequency() {
			$defaults = array(
				'homepage'   => 'always',
				'blog'       => 'daily',
				'sitemap'    => 'hourly',
				'post'       => 'weekly',
				'archive'    => 'monthly',
				'author'     => 'weekly',
				'taxonomies' => 'monthly',
			);`
